File: Web/ClientApp/src/app/domain/entities/customer.ts
import { Car, Employee } from ".";

export class Customer {
    public id: string;
    public name : string;
    public address : string;
    public phone : string;
    public passport: string;
    public orderDate: Date;
    public saleDate: Date;
    public isDone: boolean;
    public isPayment: boolean;
    public prepaymentPercentage: number;
    
    public carId: number | null;
    public car: Car;
    
    public employeeId: number | null;
    public employee: Employee;

    public static fromJS(data: any = {}): Customer {
        const entity = new Customer();
        
        entity.id = data.id;
        entity.name = data.name;
        entity.address = data.address;
        entity.phone = data.phone;
        entity.passport = data.passport;
        entity.orderDate = data.orderDate;
        entity.saleDate = data.saleDate;
        entity.isDone = data.isDone;
        entity.isPayment = data.isPayment;
        entity.prepaymentPercentage = data.prepaymentPercentage;
        entity.carId = data.carId;
        entity.car = data.car;
        entity.employeeId = data.employeeId;
        entity.employee = data.employee;

        return entity;
    }
}

File: Web/Models/Customer.cs
using System;

namespace Web.Models
{
    public class Customer
    {
        public Guid Id { get; set; }
        public string Name { get; set; } 
        public string Address { get; set; } 
        public string Phone { get; set; } 
        public string Passport { get; set; }
        public DateTime OrderDate { get; set; }
        public DateTime SaleDate { get; set; }
        public bool IsDone { get; set; }
        public bool IsPayment { get; set; }
        public double PrepaymentPercentage { get; set; }
        
        
        public int? CarId { get; set; }
        public Car Car { get; set; }
        
        public int? EmployeeId { get; set; }
        public Employee Employee { get; set; }
    }
}